### PR TITLE
Task/283/ensure all pipelines succeed

### PR DESCRIPTION
**Pull request by cyberbikepunk**
@akariv 

This processor is a replacement for the `stream_from_path` processor, whose functional approach became somewhat unreadable and brittle to changes. This new object-oriented approach should improve readability and facilitate the handling of special cases through sub-classing.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+chardet
```
Comment (akariv) [inline requirements.txt:1]: why not cchardet?
from our experience it provides better results (and much quickly)

**Comment by cyberbikepunk** [comment]
@akariv I'm not merging this PR until I've written unit-tests.

File: common/processors/ingest_local_file.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+                keys.update(set(row))
+
+        return sorted(keys)
+
+
+class XLSIngestor(BaseIngestor):
+    """An ingestor for xls and xlsx files."""
+
+    @staticmethod
+    def force_strings(rows):
```
Comment (akariv) [inline common/processors/ingest_local_file.py:247]: is someone using this method?
also:
what about XLSXIngestor?

**Review by akariv — COMMENTED** [review]
The code looks okay, although there's some overlap with what tabulator does.

File: requirements.txt (hunk 1 above)
Comment (cyberbikepunk) [reply requirements.txt:1]: thanks for the tip

File: common/processors/ingest_local_file.py (hunk 2 above)
Comment (cyberbikepunk) [reply common/processors/ingest_local_file.py:247]: I'm writing them as i need them. So far, I haven't had a use case. I'm not outright replacing the old module. I'm introducing it gradually for new and broken pipelines.

**Comment by cyberbikepunk** [comment]
@akariv @roll yes there is a lot of overlap with tabulator. It's a little unsatisfactory. But I turned the problem around and couldn't come up with a more integrated solution. For example: opening the file, cleaning the headers, and then re-opening to read the data is far from ideal.

**Comment by pwalsh** [comment]
@cyberbikepunk @akariv @roll this basically does stuff tabulator does with much much less handling of corner cases. I can't see how it was better to put the time in writing this, than it is to learn the tabulator API and/or do some small PRs to it if there was something missing (based on what I see here, there is nothing that can't be done in tabulator).

**Comment by roll** [comment]
@cyberbikepunk 
So what's challenges you've faced using processors pattern?

``` python
def processor(extended_rows):
    for number, headers, row in extended_rows:
        # clean headers
        # skip bad rows
        # ...
        # make keyed row
        yield keyed_row

def rows():
    return Stream(source, post_parse=[processor])
```

**Comment by pwalsh** [comment]
@roll small typo :)

```
def rows():
    return Stream(source, post_parse=[processor])
```

**Comment by cyberbikepunk** [comment]
Travis is not happy about changes in [os-types](https://github.com/openspending/os-types/commit/4e83e5a7b62a69b5efce77fcab1426b8969f46a7). See issue #297.

**Pull request merged**